FlowCal/excel_ui.py

```diff
@@ -216,150 +223,181 @@ def process_beads_table(beads_table,
-            density_params['ylog'] = bool(
-                beads_sample_gated.amplification_type(sc_channels[1])[0])
-            density_params["title"] = "{} ({:.1f}% retained)".format(
-                beads_id,
-                beads_sample_gated.shape[0] * 100. / beads_sample.shape[0])
-            # Plot
-            figname = os.path.join(base_dir,
-                                   plot_dir,
-                                   "density_hist_{}.png".format(beads_id))
-            plt.figure(figsize = (6,4))
-            FlowCal.plot.density_and_hist(
-                beads_sample,
-                beads_sample_gated, 
-                density_channels=sc_channels,
-                hist_channels=cluster_channels,
-                gate_contour=gate_contour, 
-                density_params=density_params,
-                hist_params={'div': 4},
-                savefig=figname)
-
-        # Save sample
-        beads_samples.append(beads_sample_gated)
-
-        ###
-        # Process MEF values
-        ###
-        # For each fluorescence channel, check whether a list of known MEF
-        # values of the bead subpopulations is provided in `beads_row`. This
-        # involves checking that a column named "[channel] MEF Values" exists
-        # and is not empty. If so, store the name of the channel in
-        # `mef_channels`, and the specified MEF values in `mef_values`.
-        ###
-        mef_values = []
-        mef_channels = []
-        for fl_channel in fl_channels:
-            if fl_channel in mef_channels_all:
-                # Get header from channel name
-                mef_header = mef_headers_all[mef_channels_all.index(fl_channel)]
-                # Extract text. If empty, ignore.
-                mef_str = beads_row[mef_header]
-                if pd.isnull(mef_str):
-                    continue
-                # Save channel name
-                mef_channels.append(fl_channel)
-                # Parse list of values
-                mef = mef_str.split(',')
-                mef = [int(e) if e.strip().isdigit() else np.nan
-                       for e in mef]
-                mef_values.append(mef)
-        mef_values = np.array(mef_values)
-
-        # Obtain standard curve transformation
-        if mef_channels:
+                print("\nBeads ID {}...".format(beads_id))
+                print("Loading file \"{}\"...".format(beads_row['File Path']))
+
+            # Attempt to open file
+            filename = os.path.join(base_dir, beads_row['File Path'])
+            try:
+                beads_sample = FlowCal.io.FCSData(filename)
+            except:
+                raise ExcelUIException("file \"{}\" not found".format(
+                    beads_row['File Path']))
+            # Check that the number of events is greater than 500
+            if beads_sample.shape[0] < 500:
+                raise ExcelUIException("number of events is lower than 500")
```

Comment:
My approach has been to just catch the most common errors. The point of this error check in particular is to make sure that the FCS file has enough events for all the gating steps downstream. Having files with low event counts is apparently something people do quite frequently for some reason.

Alternatively, we could wrap the gating function in a `try..except`, and throw an `ExcelUIException` with a more user-friendly message. This could be done for all gating functions. However, I find it easier to just check for a defined number of events at the beginning and say that we don't support files with less events than this number.

> I think we should just defer to any error messages that are raised by any downstream processes that fail (which should ideally give a more detailed explanation for why there were too few events).

So do you think we should have a general `except` statement at the end of the iteration that captures any error that we haven't though of, and prints it to the Excel UI? I'm not against that, but it is not what I was trying to do.